src/modules/select/wallets/keepkey/pinModal.ts

```diff
@@ -0,0 +1,178 @@
+import Button from '../../../../elements/Button.svelte'
+import { detach, insert, noop } from 'svelte/internal'
+
+const buttonStyles = `
+  align-items: center;
+  border-radius: 6px;
+  border: none;
+  cursor: pointer;
+  display: flex;
+  justify-content: center;
+  line-height: 1;
+  font-size: 18px;
+`
+
+const HANDLE_PIN_PRESS = 'handlePinPress'
+
+const pinButton = (
+  value: number,
+  slot?: string,
+  width = '46px',
+  height = '46px'
+) => `
+  <button
+    id="pin-button-${value}"
+    style="${buttonStyles} width: ${width}; height: ${height};"
+    type="button"
+    onclick="window.${HANDLE_PIN_PRESS}(${value})">
+      ${
+        slot ||
+        `<svg viewBox="0 0 24 24" width="16" height="16">
+        <circle cx="12" cy="12" r="10" fill="#555555"></circle>
+      </svg>`
+      }
+  </button>
+`
+const pinPadStyles = `
+  display: grid;
+  grid-template-columns: repeat(3, 65px);
+  grid-template-rows: repeat(3, 65px);
+  align-items: center;
+  justify-items: center;
+`
+
+const pinButtons = `
+  <div style="${pinPadStyles}">
+    ${[7, 8, 9, 4, 5, 6, 1, 2, 3].map(val => pinButton(val)).join('')}
+  </div>
+`
+
+const pinInputStyles = `
+  background: inherit;
+  font-size: 0.889em;
+  font-family: inherit;
+  border-width: 1px;
+  border-style: solid;
+  border-color: inherit;
+  border-radius: 4px;
+  padding-left: 0.5rem;
+  padding-right: 4.1rem;
+  color: inherit;
+  transition: background 150ms ease-in-out;
+  height: 32px;
+`
+
+const pinInputContainerStyles = `
+  margin: 15px 0;
+  display: flex;
+`
+
+const pinInput = `
+  <div style="${pinInputContainerStyles}">
+    <input
+      id="pin-input"
+      style="${pinInputStyles}"
+      placeholder="PIN"
+      type="password"
+    />
+    <div style="margin-left: 5px">
+      ${pinButton(-1, '❮', '36px', '36px')}
+    </div>
+  </div>
+`
+const pinPadContainerStyles = `
+  align-items: center;
+  display: flex;
+  flex-flow: column;
+  padding: 20px 10px;
+`
+
+export const renderPinModal = (wallet: KeepKeyHDWallet) => {
+  const pinPadHTML = `
+    <style>.bn-branding { visibility: hidden !important; }</style>
+    <div id="pin-pad" style="${pinPadContainerStyles}" class="bn-onboard-custom">
+      ${pinButtons}
+      ${pinInput}
+    </div>
+  `
+
+  const getPinInput = () =>
+    document.getElementById('pin-input') as HTMLInputElement
+
+  const handlePinPress = (value: number) => {
+    const input = getPinInput()
+    // A value of -1 signals a backspace e.g. we delete the last char from the input
+    input.value = value === -1 ? input.value.slice(0, -1) : input.value + value
+  }
+
+  const deleteWindowProperties = () => {
+    delete (window as any)[HANDLE_PIN_PRESS]
+  }
+
+  ;(window as any)[HANDLE_PIN_PRESS] = handlePinPress
+
+  // Creates a modal component which gets mounted to the body and is passed the pin html into it's slot
+  const div = document.createElement('div')
+  div.innerHTML = pinPadHTML
+  const pinModal = new Modal({
```

Comment:
Using the component constructor made it possible to reuse not only the Modal/Button styles but the logic as well